Automattic/sensei
# Implement backend for Help section in Sensei Home.

Fixes #5742 

### Changes proposed in this Pull Request
* Implement backend for Help section.
* Added `sensei_home_support_ticket_creation_upsell_show` filter.

### Testing instructions
* Run `curl -sX GET --user aaron:development --insecure 'https://test2.local/wp-json/sensei-internal/v1/home' | jq .help` and confirm that output is as expected. ([Check how to test curl in local more easily](https://evergreenp2.wordpress.com/2022/03/03/testing-cheatsheet/#test-rest-api-in-local-easily))
* Disbale `sensei_home_support_ticket_creation_upsell_show` with a snippet.
* Run the same curl from above and confirm that output changed correctly.

### New hooks/filters
- Added `sensei_home_support_ticket_creation_upsell_show` filter to allow to disable upsell action to Sensei Pro.

### Screenshot / Video
**Default:**
```
$ curl -sX GET --user aaron:development --insecure 'https://test2.local/wp-json/sensei-internal/v1/home' | jq .help
[
  {
    "title": "Get the most out of Sensei",
    "items": [
      {
        "title": "Sensei Documentation",
        "url": "https://senseilms.com/documentation/",
        "icon": null,
        "extra_link": null
      },
      {
        "title": "Support forums",
        "url": "https://wordpress.org/support/plugin/sensei-lms/",
        "icon": null,
        "extra_link": null
      },
      {
        "title": "Create a support ticket",
        "url": null,
        "icon": "lock",
        "extra_link": {
          "label": "Upgrade to Sensei Pro",
          "url": "https://senseilms.com/pricing/"
        }
      }
    ]
  }
]
```

**With `sensei_home_support_ticket_creation_upsell_show` disabled:**
```
$ curl -sX GET --user aaron:development --insecure 'https://test2.local/wp-json/sensei-internal/v1/home' | jq .help
[
  {
    "title": "Get the most out of Sensei",
    "items": [
      {
        "title": "Sensei Documentation",
        "url": "https://senseilms.com/documentation/",
        "icon": null,
        "extra_link": null
      },
      {
        "title": "Support forums",
        "url": "https://wordpress.org/support/plugin/sensei-lms/",
        "icon": null,
        "extra_link": null
      },
      {
        "title": "Create a support ticket",
        "url": "https://senseilms.com/contact/",
        "icon": "lock",
        "extra_link": null
      }
    ]
  }
]
```
